### PR TITLE
Feature - store RA in data

### PR DESCRIPTION
**Pull request by joefitter**
* added raDate to project if applic on grant, 6 months after expiry
* added ra props to version on grant
* set raDate as 6 months after revocation on revoke

### CODE REVIEW

File: lib/resolvers/project-version.js [hunk 1]
```diff
@@ -51,8 +35,27 @@ module.exports = ({ models }) => ({ action, data, id }, transaction) => {
       });
   }
 
-  if (action === 'update' || action === 'patch') {
-    return updateVersion()
+  if (action === 'patch') {
```
Comment (lennym) [inline lib/resolvers/project-version.js:38]: Are we _absolutely_ sure there's no lingering `update`s?

File: lib/resolvers/project.js [hunk 2]
```diff
@@ -202,6 +210,8 @@ module.exports = ({ models }) => async ({ action, data, id, meta = {} }, transac
       patch.expiryDate = start.add({ months, years }).toISOString();
     }
 
+    patch.raDate = getRaDate(versionToGrant, patch.expiryDate || versionToGrant.expiryDate);
```
Comment (lennym) [inline lib/resolvers/project.js:213]: The expiry date will be on the project, not the version, right?
```suggestion
    patch.raDate = getRaDate(versionToGrant, patch.expiryDate || project.expiryDate);
```

File: lib/resolvers/project-version.js (hunk 1 above)
Comment (joefitter) [reply lib/resolvers/project-version.js:38]: I would be surprised as I'm 99% sure they autoresolved, however we could leave it to be safe?

File: lib/resolvers/project.js (hunk 2 above)
Comment (joefitter) [reply lib/resolvers/project.js:213]: ooh good spot 👍

**Pull request merged**